Starter Tutorial

FernandoAscencio opened the pull request:
### Main
#### Implemented a basic Tutorial
Things you can check:
- Upon starting the app for the first time, the tutorial should run automatically.
- Tutorial should take you through all four Activities: Home, Camera, Live Translation, and Settings
- Once ended, tutorial should never start again, even when closing and opening the app.
- You can go to Settings and click on the "Tutorial" option. It should start the tutorial from the Main/Home page.

Future Additions:
- Looking to implement a Tutorial Class
- Adding all the Strings from the Tutorial to the string.xml resource for locale-config. i.e. language changes.

### Extra
- Added Strings for locale-config from the Preferences/Settings menu among others.
- Some Formatting changes.

P.S.: I should probably have kept the Main and Extra as separate commits. Sorry.

FernandoAscencio committed:
Start of Tutorial

Added Strings for locale-config
- Preferences

mycoal2 reviewed (dismissed):
ran on my device, looks good to me.

m-sepehr committed:
fixed bug where NullPointerException occurred in onImageAvailable in CameraActivity.java during Tutorial, causing app to crash

m-sepehr committed:
fixed bug by preventing IllegalStateException

In CameraConnectionFragment.java, the createCameraPreviewSession method could result in an IllegalStateException ("CameraDevice was already closed") if the camera device was closed before creating the capture session. To fix, added a null check to ensure that the cameraDevice is not null before attempting to create the capture session, preventing the exception.

FernandoAscencio committed:
Rebase corrections

FernandoAscencio committed:
Rebase Corrections

- Add Homepage buttons to Tutorial
- Unifies toolbar color for Activities
- Camera item id and name changed from "page_1" to "camera"
- Titles moved to string.xml for locale-config languages.

FernandoAscencio committed:
Small clean up

Replacing with lambdas.
